polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp

```diff
@@ -648,58 +648,51 @@ void LoopInternalization::runOnOperation() {
 
   // Collect kernel body functions of candidate kernels.
   std::set<FunctionOpInterface> kernelBodyFuncs;
-  std::set<FunctionOpInterface> toRemove;
   gpuModule->walk([&](gpu::GPUFuncOp kernel) {
-    if (!kernel.isKernel())
+    if (!kernel.isKernel() || !isCandidateKernel(kernel))
       return;
-    FunctionOpInterface func = funcKernelInfo.getKernelBodyFunc(kernel);
-    if (isCandidateKernel(kernel))
-      kernelBodyFuncs.insert(func);
-    else
-      toRemove.insert(func);
-  });
 
-  // If func is a kernel body function of a non-candidate kernel, then remove it
-  // from the set.
-  for (FunctionOpInterface func : toRemove)
-    kernelBodyFuncs.erase(func);
+    std::set<FunctionOpInterface> kernelBodyFunctions =
```

Comment:
If there is a kernel body function `func`, called by `kernel1` and `kernel2`, where `kernel1` is a candidate, and `kernel2` is not a candidate. This change would have added `func` in `kernelBodyFuncs`. Can we avoid that from happening, or assert that it doesn't happen?

Reply:
ok, wanted to assert to ensure our expectation is correct, but no need if it adds complexity.